src/transitions.js

```diff
@@ -0,0 +1,250 @@
+    const transformCache = el.style.transform
+    const transformOriginCache = el.style.transformOrigin
+
+    // Default values inspired by: https://material.io/design/motion/speed.html#duration
+    const styleValues = {
+        duration: display || (transition.in && transition.out)
+            ? modifierValue(modifiers, 'duration', 150)
+            : modifierValue(modifiers, 'duration', 150) / 2,
+        origin: modifierValue(modifiers, 'origin', 'center'),
+        first: {
+            opacity: display ? 0 : 1,
+            scale: display ? modifierValue(modifiers, 'scale', 95) : 100,
+        },
+        second: {
+            opacity: display ? 1 : 0,
+            scale: display ? 100 : modifierValue(modifiers, 'scale', 95),
+        },
+    }
+
+    // These are the explicit stages of a transition (same stages for in and for out).
+    // This way you can get a birds eye view of the hooks, and the differences
+    // between them.
+    const stages = {
+        start() {
+            if (transitionOpacity) el.style.opacity = styleValues.first.opacity
+            if (transitionScale) el.style.transform = `scale(${styleValues.first.scale / 100})`
+        },
+        during() {
+            if (transitionScale) el.style.transformOrigin = styleValues.origin
+            el.style.transitionProperty = [(transitionOpacity ? `opacity` : ``), (transitionScale ? `transform` : ``)].join(' ').trim()
+            el.style.transitionDuration = `${styleValues.duration / 1000}s`
+            el.style.transitionTimingFunction = `cubic-bezier(0.4, 0.0, 0.2, 1)`
+        },
+        show() {
+            // Resolve if showing
+            if (display) resolve()
+        },
+        end() {
+            if (transitionOpacity) el.style.opacity = styleValues.second.opacity
+            if (transitionScale) el.style.transform = `scale(${styleValues.second.scale / 100})`
+        },
+        hide() {
+            // Resolve if hiding
+            if (! display) resolve()
+        },
+        cleanup() {
+            if (transitionOpacity) el.style.opacity = opacityCache
+            if (transitionScale) el.style.transform = transformCache
+            if (transitionScale) el.style.transformOrigin = transformOriginCache
+            el.style.transitionProperty = null
+            el.style.transitionDuration = null
+            el.style.transitionTimingFunction = null
+        },
+    }
+    // Render transition with inline styles
+    renderStages(el, stages)
+}
+
+function transitionWithCssClasses(el, component, resolve, attrs, transition, display) {
+    const originalClasses = el.__x_original_classes || []
+
+    let ensureStringExpression = (expression) => {
+        return typeof expression === 'function'
+            ? component.evaluateReturnExpression(el, expression)
+            : expression
+    }
+
+    // Prepare stage group names for given directions
+    let cssClasses = {
+        durring: transition,
```

Comment:
Is this a typo?

Reply:
Which? one,`cssClasses`? 

No, down on the road it will serve as css class container. The description was for the pre assigned stuff. Basically css classes assigned by their stage group names. So at this point we were preparing the container.

Reply:
I think Hugo was referring to the word `durring`

Reply:
oh it is a typo then.